### PR TITLE
New Worksheet Generator Enhancement

### PR DESCRIPTION
**Pull request by AaronSosaRamos**
This PR was made for improving the internal Worksheet Generator workflow removing the `question_type` and `number_of_questions` args. and optimizing it with the LLM enhancement on this.  New payloads are updated in #114

### CODE REVIEW

File: app/features/worksheet_generator/core.py [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
-    
     course_type = generate_course_type(topic, verbose)['course_type']
 
     if verbose:
         logger.info(f"Course Type: {course_type} [{lang}]")
         logger.info(f"File URL loaded: {file_url}")
 
     docs = get_docs(file_url, file_type, verbose)
+
+    worksheet_list = worksheet_question_type_generator(course_type=course_type,
```
Comment (yunusj) [inline app/features/worksheet_generator/core.py:27]: @WilfredoAaronSosaRamosBoushtech this tool has 3 LLM calls with this refactoring now. I was wondering is there a way to combine this in a single call or perhaps you can help me understand your reasoning

Comment (AaronSosaRamos) [reply app/features/worksheet_generator/core.py:27]: @yunusj I've done 3 LLM calls following this internal process:
1. The first LLM call is done in order to identify the course type based in the International Baccaulereate guidelines (For adapting the LLM understanding of the nature, methodology and body of knowledge of the course types):
https://www.ibo.org/programmes/diploma-programme/curriculum/
2. The second LLM call is to identify what question types and number of questions are required for the given inputs.
3. The third LLM call is for generating the worksheet.

Comment (AaronSosaRamos) [reply app/features/worksheet_generator/core.py:27]: I used multiple shot prompting for adapting the generated Worksheet in base of the nature and methodology of the course type based in the International Baccaulereate.

**Comment by AaronSosaRamos** [comment]
I've refactored the Worksheet Generator using multithreading support and moving the document loaders to a global utils folder for better reusability.

**Review by yunusj — APPROVED** [review]
Looks good. Are the arguments/inputs for this tool changed? If so, we'll need to update our frontend integration as well

**Pull request merged**